docs/examples/testing.md

```diff
@@ -0,0 +1,144 @@
+---
+description: |
+  By creating an application handler, you can write tests.
+---
+
+We can be create an application handler from `createHandler()` and use develop
+for testing HTTP.
+
+An example of testing three URLs and two HTTP request methods is as follows
+
+### handlers source
+
+```tsx
+// routes/index.tsx
+
+import { HandlerContext, Handlers } from "$fresh/server.ts";
+
+export const handler: Handlers = {
+  async POST(req: Request, ctx: HandlerContext) {
+    const form = await req.formData();
+
+    // Processing something
+
+    return new Response("", {
+      status: 303,
+      headers: { Location: "/" },
+    });
+  },
+};
+
+export default function Index() {
+  return (
+    <div>
+      Hello Deno!
+    </div>
+  );
+}
+```
+
+```tsx
+// routes/foo.tsx
+
+export default function Foo() {
+  return (
+    <div>
+      Hello Foo!
+    </div>
+  );
+}
+```
+
+### Test code
+
+```ts
+// examples\test\test\main_test.ts
+
+import { createHandler } from "$fresh/server.ts";
+import manifest from "../fresh.gen.ts";
+import { assertEquals, assertExists } from "$std/testing/asserts.ts";
+import type { ConnInfo } from "../../../src/server/deps.ts";
+
+const CONN_INFO: ConnInfo = {
+  localAddr: { hostname: "127.0.0.1", port: 8000, transport: "tcp" },
+  remoteAddr: { hostname: "127.0.0.1", port: 53496, transport: "tcp" },
+};
+
+Deno.test("HTTP assert test.", async (t) => {
+  await t.step("#1 GET /", async () => {
+    const handler = await createHandler(manifest);
```

Comment:
`handler` can be moved to the start of the test and only declared once, instead at every test step.